constellation-app/constellation
# Added plugin info from Data Access View plugins to plugin reporter

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
Added time, parameters (if any), and selected items (nodes/transactions) if any, to the description sent to the plugin reporter from Data Access View plugins.  Also added all the messages to Save Details to Clipboard.

Added overloaded `SetProgress` method in `DefaultPluginInteractions` to be able to add `parameters` and `selected` as needed, so as to be able to extract the data to add to messages.

Then for plugins in Data Access View that had parameters and/or selection, updated the `setProgress` to pass in the relevant parameters. I usually only the updated the second `setProgress` method in the plugin edit/run method, as the first one is usually the message of starting progress and is before the parameters are obtained. Note that parameters and selected are added only once as you don't want the parameters output for every message.



### Alternate Designs



### Why Should This Be In Core?
This should be available to all current plugins and future plugins if the parameters and selection provide more information in messages to better understand the data output.



### Benefits
To provide more information in messages to better understand the data output.


### Possible Drawbacks



### Verification Process

1. Run Constellation
2. Open a graph
3. Open Data Access View
4. Test with the various plugins

Time, parameters and in some cases selected nodes/transactions may be seen in the plugin reporter.

5. Right-click on the message and select "Save details To Clipboard", then paste the clipboard somewhere to read. The parameters and selection should be in there.



### Applicable Issues
#49
